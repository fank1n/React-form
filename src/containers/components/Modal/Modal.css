.modal {
    height: 100vh;
    width: 100vw;
    background-color: rgba(0, 0, 0, 0.4);
    position: fixed;
    top: 0;
    left: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    opacity: 0;
    transition: 0.5s;
    pointer-events: none;
  }
  .modal.active {
    opacity: 1;
    pointer-events: all;
    /* word-break: inherit; */
  }
  .modal-content {
    padding: 20px 20px 80px;
    border-radius: 5px;
    background-color: #fff;
    height: 200px;
    width: 400px;
    transform: scale(0.5);
    transition: 0.3s all;
  }
  .modal-content.active{
      transform: scale(1);
  }