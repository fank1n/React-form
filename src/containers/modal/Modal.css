.modal {
  height: 100vh;
  width: 100vw;
  background-color: rgba(0, 0, 0, 0.4);
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  opacity: 0;
  transition: 0.5s;
  pointer-events: none;
}

.modal.active {
  opacity: 1;
  pointer-events: all;
}
.modal-content {
  padding: 20px 20px 80px;
  border-radius: 5px;
  background-color: #fff;
  height: 200px;
  width: 400px;
  scale: (0.5);
  transition: 0.3s;
}
.modal-content.active {
  transform: scale(1);
}
.modal-name {
  margin-bottom: 0;
  text-align: center;
  margin-top: 0;
  font-size: 24px;
}
.modal-inputs-container {
  text-align: center;
}
.modal-inputs {
  margin: 0 auto;
}
.modal-inputs:nth-child(4) {
  margin-bottom: 20px;
}
.paragraph {
  margin: 10px;
  margin-top: 20px;
}
#modal-btn {
  background-color: white;
  color: rgb(36, 36, 36);
  border: 2px solid rgb(36, 36, 36);
  width: 120px;
  height: 40px;
  border-radius: 3px;
  cursor: pointer;
  font-size: 14px;
  margin-top: 20px;
}
#modal-btn:disabled {
  background-color: #ccc;
  color: rgb(134, 134, 134);
  border: none;
}
#modal-btn:disabled:hover {
  background-color: #ccc;
  color: rgb(134, 134, 134);
  cursor: auto;
}
#modal-btn:hover {
  color: white;
  background-color: rgb(36, 36, 36);
}
.modal-cross {
  margin-left: 380px;
  cursor: pointer;
}
.modal-errors {
  color: red;
  margin: 0;
  font-size: 13px;
}