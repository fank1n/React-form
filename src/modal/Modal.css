.modal {
  height: 100vh;
  width: 100vw;
  background-color: rgba(0, 0, 0, 0.4);
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  opacity: 0;
  transition: 0.5s;
  pointer-events: none;
}

.modal.active {
  opacity: 1;
  pointer-events: all;
}
.modal-content {
  padding: 20px;
  border-radius: 5px;
  background-color: #fff;
  height: 200px;
  width: 400px;
  scale: (0.5);
  transition: 0.3s;
}
.modal-content.active {
  transform: scale(1);
}
.modal-name {
  text-align: center;
  margin-top: 0;
  font-size: 24px;
}
.modal-inputs-container {
  text-align: center;
}
.modal-inputs {
  margin: 0 auto;
}
.modal-inputs:nth-child(4) {
  margin-bottom: 10px;
}
.paragraph {
  margin: 10px;
}
#modal-btn {
  background-color: rgb(36, 36, 36);
  color: white;
  border: none;
  width: 120px;
  height: 30px;
  border-radius: 3px;
  cursor: pointer;
}