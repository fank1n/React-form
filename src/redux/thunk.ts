// import {getTodos} from "../api/todos";
import INITIALIZE_TODOS from "./actions/todoActions"

export interface ListItem {
    "id": number,
    "description": string,
    "completed": false,
    "favorite": false
}

export const getTodos = () => {
    return async (dispatch:any) => {
        const todos = await fetch ("http://localhost:8000/todo")
        .then((todos) => todos.json)
        .then((todos) => {
            dispatch({
                type: INITIALIZE_TODOS,
                payload: todos,
            })
        }).catch((error) => console.log(error))
    }
}

