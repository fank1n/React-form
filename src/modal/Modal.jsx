import "./Modal.css";
import { useState, useContext, useEffect } from "react";
import Context from "../context";

const Modal = ({ active, setActive }) => {
  const {loggedInfo, setFormValid, setIsAdmin} = useContext(Context);
  const [login, setLogin] = useState('');
  const [password, setPassword] = useState('');
  const [isLoginClicked, setIsLoginClicked] = useState(false)
  const [isPasswordClicked, setIsPasswordClicked] = useState(false)
  const [loginError, setLoginError] = useState('Некорректный логин')
  const [passwordError, setPasswordError] = useState('Некорректный пароль')
  

  useEffect(() => {
    if(loginError || passwordError){
      setFormValid(false)
    }
    else{
      setActive(false)
      setFormValid(true);
    }
  }, [loginError, passwordError])

  const blurHandler = (e) => {
    switch (e.target.name) {
      case 'login':
          setIsLoginClicked(true)
        break
    
      case 'password':
        setIsPasswordClicked(true)
        break
    }
  }
  const loginHandler = (e) => {
    setLogin(e.target.value)
    const trueAdminLogin = loggedInfo.userData[0].login;
    const trueCustomerLogin = loggedInfo.userData[1].login;

    if(e.target.value === trueAdminLogin){
      setLoginError('');
      setIsAdmin(true);
    }
    else if (e.target.value === trueCustomerLogin){
      setLoginError('');
      setIsAdmin(false);
    }
    else{
      setLoginError('Некорректно введенный логин')
      setIsAdmin(false);
    }
  }
  const passwordHandler = (e) => {
    setPassword(e.target.value)
    const trueAdminPassword = loggedInfo.userData[0].password;
    const trueCustomerPassword = loggedInfo.userData[1].password;
    
    if(e.target.value === trueAdminPassword){
      setPasswordError('');
      setIsAdmin(true);
    }
    else if(e.target.value === trueCustomerPassword){
      setPasswordError('');
      setIsAdmin(false);
    }
    else{
      setPasswordError('Некорректно введенный пароль');
      setIsAdmin(false);
    }
  }

  return (
    <div
      className={active ? "modal active " : "modal"}
      onClick={() => setActive(false)}
    >
      <div
        className={active ? "modal-content active " : "modal-content"}
        onClick={(e) => {
          e.stopPropagation();
          e.preventDefault();
        }}
      >
        <div>
          <img
            className="modal-cross"
            src="https://cdn-icons-png.flaticon.com/512/659/659891.png"
            height="20px"
            width="20px"
            onClick={() => {
              setActive(false);
            }}
          ></img>
          <p className="modal-name">Авторизация</p>
        </div>
        <form className="modal-inputs-container">
          <p className="paragraph">Логин</p>
          <input 
          className="modal-inputs" 
          value = {login} 
          name = "login" 
          type="text"
          onBlur = {e => blurHandler(e)}
          onChange = {e => loginHandler(e)}
          >
          </input>
          {(isLoginClicked && loginError) && <p className = "modal-errors">{loginError}</p>}
          <p className="paragraph">Пароль</p>
          <input 
          className="modal-inputs"
           value = {password} 
           name = "password" 
           type="password"
           onBlur = {e => blurHandler(e)}
           onChange = {e => passwordHandler(e)}
           >
           </input>
           {(isPasswordClicked && passwordError) && <p className = "modal-errors">{passwordError}</p>}
        </form>
      </div>
    </div>
  );
};

export default Modal;
