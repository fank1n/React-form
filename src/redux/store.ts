import { applyMiddleware, createStore, compose } from 'redux';
// import { configureStore } from "@reduxjs/toolkit";
import todosReducer from "./reducers/addTodoReducer";
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
// export const store = configureStore({
//   reducer: todosReducer,
// });

// export const store = createStore(
//   todosReducer, 
//   compose(applyMiddleware(ReduxThunk),     
//   window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())
// );
// export const store = createStore(
//   todosReducer,
//   compose(
//       applyMiddleware(thunk),
//   )
// )
export const store = createStore(todosReducer, composeWithDevTools(applyMiddleware(thunk)));

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;