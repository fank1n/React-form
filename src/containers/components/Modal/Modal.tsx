import React from "react";
import { useDispatch } from "react-redux";
import { removeTodo } from "../../../redux/reducers/addTodoReducer";
import { AppDispatch } from "../../../redux/store";
import "./Modal.css";
import cross from "../../../icons/cross.svg"
const Modal = ({postDate, modalText, active, setActive, id }:any) => {
    const dispatch = useDispatch<AppDispatch>();
    
    const handleDeleteTodo = () => {
        dispatch(removeTodo(id));
        setActive(!active);
    }

    return (
        <div className = {active ? "modal active" : "modal"} onClick = {() => setActive(!active)}>
            <div className={active ? "modal-content active" : "modal-content"} onClick = {e => e.stopPropagation()}>
                <img 
                className = "modal-close-cross"
                src = {cross} 
                alt = "cross"
                onClick = {() => setActive(false)}
                />
                <h2 className = "modal-title">Вы действительно хотите удалить задачу?</h2>
                <div className="modal-text-container">
                    <p className = "modalText">{modalText}</p>
                </div>
                <p className = "modal-date">Дата создания: {postDate}</p>
                <div className="modal-btn-container">
                    <button className="modal-cancel-btn" data-test-id="cancel-btn" onClick={() => setActive(false)}>Отмена</button>
                    <button className="modal-confirm-btn" onClick = {handleDeleteTodo}>Да, удалить</button>
                </div>
            </div>
        </div>
    );
}
export default Modal;