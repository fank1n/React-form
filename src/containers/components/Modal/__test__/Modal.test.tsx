// import React from 'react';
import Modal from '../Modal';
import {describe, expect} from '@jest/globals'
import {mount} from "enzyme"
import { Provider } from 'react-redux';
import { store } from "../../../../redux/store"

describe('Modal', () => {
    let wrapper:any
    let activeStatus = true
    const mockfunc = jest.fn()
    beforeEach(() => {
      wrapper = mount(<Provider store={store}><Modal setActive={mockfunc} active={activeStatus}/></Provider>)
    });

    afterEach(() => {
      wrapper.unmount()
    });
    
    it('Modal component snapshot should be correct', () => {
      expect(wrapper).toMatchSnapshot()
    });

    it('Should close modal on close(cross) button', () => {
      wrapper.find('[className="modal-close-cross"]').simulate('click');
      expect(mockfunc).toBeCalledTimes(1)
    });

    it('Should close modal on close(Отмена) button', () => {
      wrapper.find('[data-test-id="cancel-btn"]').simulate('click');
      expect(mockfunc).toBeCalledTimes(1)
    });

    it('Modal component should render elements currectly', () => {
      expect(wrapper.find('[className="modal active"]')).toBeTruthy()
      expect(wrapper.find('[className="modal-close-cross"]')).toBeTruthy()
      expect(wrapper.find('[className="modal-cancel-btn"]')).toBeTruthy()
      expect(wrapper.find('[className="modal-confirm-btn"]')).toBeTruthy()
    });
  });

