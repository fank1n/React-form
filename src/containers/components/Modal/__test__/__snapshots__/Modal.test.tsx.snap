// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Modal component should match shapshot 1`] = `ReactWrapper {}`;

exports[`Modal Modal component should render correctly 1`] = `ReactWrapper {}`;

exports[`Modal Modal component should render elements currectly 1`] = `ReactWrapper {}`;

exports[`Modal Modal component snapshot should be correct 1`] = `ReactWrapper {}`;
